### DISCUSSION TITLE
Inconsistent behavior of navigation with manual Suspense boundaries

### DISCUSSION THREAD

**Discussion by SidKH** [vercel/next.js#49377] [App Router]
The navigation respects suspense fallback locally but ignores it in prod. 

Simplified code of the page 👇

```jsx
export default async function MoviePage({params}) {
  const selectedMovie = await getMovie(params.id);
  return (
    <div>
      <MovieHeader movie={selectedMovie} />
      <Suspense fallback={<>Loading...</>}>
        <SimilarMovies movie={selectedMovie} />
      </Suspense>
   </div>
  )
}
```

Locally, navigation works as expected.

The new route shows `<MovieHeader />` and the fallback while `<SimilarMovies />` fetching data (GPT response). When the response is ready, `<SimilarMovies />` gets rendered.  

https://user-images.githubusercontent.com/7762253/236639408-9cfdd907-fa6f-4743-91bf-38be436a5cae.mp4

However, in prod, it works differently. 

When I click a `Link,` the router doesn't show a suspense fallback. Instead, it waits for the data in `<SimilarMovies />` to get fetched and only then switches to the new route and renders everything all at once.

If I reload the page, it does show the suspense fallback.

https://user-images.githubusercontent.com/7762253/236639755-02441a8b-ddb2-4359-8e91-cc9944debd5b.mp4

**Comment by AntoineBourin** [comment]
Same behavior on my side but only when I add a "force-static" options in my page.js file or if I add a generateStaticParams function. 
Did you have one of these ?
